src/ripple/app/tests/ValidatorList_test.cpp

```diff
@@ -0,0 +1,228 @@
+
+        if (blob.empty())
+            LogicError ("Can't convert invalid RippleAddress to PublicKey");
+
+        return PublicKey(Slice(blob.data(), blob.size()));
+    }
+
+    static
+    std::unique_ptr<ValidatorList>
+    make_ValidatorList (
+        std::vector<RippleAddress> const& permanent,
+        std::vector<RippleAddress> const& ephemeral)
+    {
+        auto vl = std::make_unique <ValidatorList> (beast::Journal ());
+
+        for (auto const& n : permanent)
+            vl->insertPermanentKey (n, n.ToString());
+
+        for (auto const& n : ephemeral)
+            vl->insertEphemeralKey (asPublicKey (n), n.ToString());
+
+        return vl;
+    }
+
+    static
+    RippleAddress
+    randomNode ()
+    {
+        return RippleAddress::createNodePublic (
+            RippleAddress::createSeedRandom ());
+    }
+
+    static
+    bool
+    isPresent (
+        std::vector<RippleAddress> container,
+        RippleAddress const& item)
+    {
+        auto found = std::find (
+            std::begin (container),
+            std::end (container),
+            item);
+
+        return (found != std::end (container));
+    }
+
+    void
+    testMembership ()
+    {
+        // The servers on the permanentValidators
+        std::vector<RippleAddress> permanentValidators;
+        std::vector<RippleAddress> ephemeralValidators;
+
+        while (permanentValidators.size () != 64)
+            permanentValidators.push_back (randomNode());
+
+        while (ephemeralValidators.size () != 64)
+            ephemeralValidators.push_back (randomNode());
+
+        {
+            testcase ("Membership: No Validators");
+
+            auto vl = make_ValidatorList ({}, {});
+
+            for (auto const& v : permanentValidators)
+                expect (!vl->member (v));
+
+            for (auto const& v : ephemeralValidators)
+                expect (!vl->member (v));
+        }
+
+        {
+            testcase ("Membership: Non-Empty, Some Present, Some Not Present");
+
+            std::vector<RippleAddress> p (
+                permanentValidators.begin (),
+                permanentValidators.begin () + 16);
+
+            while (p.size () != 32)
+                p.push_back (randomNode());
+
+            std::vector<RippleAddress> e (
+                ephemeralValidators.begin (),
+                ephemeralValidators.begin () + 16);
+
+            while (e.size () != 32)
+                e.push_back (randomNode());
+
+            auto vl = make_ValidatorList (p, e);
+
+            for (auto const& v : p)
+                expect (vl->member (v));
+
+            for (auto const& v : e)
+                expect (vl->member (v));
+
+            for (auto const& v : permanentValidators)
+                expect (static_cast<bool>(vl->member (v)) == isPresent (p, v));
+
+            for (auto const& v : ephemeralValidators)
+                expect (static_cast<bool>(vl->member (v)) == isPresent (e, v));
```

Comment:
How about
```
for (auto it = permanentValidators.begin() + p.size(); it != permanentValidators.end(); ++it)
  expect (!vl->member(*it));

for (auto it = ephemeralValidators.begin() + e.size(); it != ephemeralValidators.end(); ++it)
  expect (!vl->member(*it));
```
It's a little uglier because you're now working directly with the iterator, but it removes the need for `isPresent`.